keps/976-plain-pods/README.md

```diff
@@ -0,0 +1,644 @@
+
+2. Pods belonging to specific namespaces (such as kube-system).
+
+The set of namespaces is defined in Configuration.Integrations
+
+```golang
+type Integrations struct {
+  Frameworks []string
+  PodOptions *PodIntegrationPolicy
+}
+
+type PodIntegrationOptions struct {
+  NamespaceSelector *metav1.LabelSelector
+}
+```
+
+When empty, Kueue uses the following NamespaceSelector internally:
+
+```yaml
+matchExpressions:
+- key: kubernetes.io/metadata.name
+  operator: NotIn
+  values: [kube-system]
+```
+
+### Constructing Workload objects
+
+Once the webhook has marked Pods subject to queuing with the `kueue.x-k8s.io/managed: true` label,
+the Pod reconciler can create the corresponding Workload object to feed the Kueue admission logic.
+
+Note that the Workload cannot be owned by the Pod. Otherwise any cascade deletion of the Pod would
+delete the Workload object, even before the Pod terminates (if it has a grace period).
+This means that we need to manually delete the Workload object once we have determined that all pods
+have finished.
+
+#### Single Pods
+
+The simplest case we want to support is single Pod jobs. These Pods only have the label
+`kueue.x-k8s.io/queue-name`, indicating the local queue where they will be queued.
+
+When constructing the Workload object, kueue only populates a single podset using the podRef field.
+The Workload for the Pod in [story 1](#story-1) would look as follows:
+
+```yaml
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: Workload
+metadata:
+  name: pod-foo
+  namespace: pod-namespace
+spec:
+  queueName: queue-name
+  podSets:
+  - count: 1
+    name: main # this name is irrelevant.
+    podRef: foo
+```
+
+<<[UNRESOLVED creating a Workload beforehand]>>
+Could users create a Workload object before hand for groups of Pods?
+This way, Pods would only need to have a label for the group name and the role. This would
+simplify validation.
+Should we make this a supported mode in addition to pure labels/annotations?
+<<[/UNRESOLVED]>>
+
+#### Groups of Pods with the same shape
+
+When multiple pods belong to the same group and have the same shape, we need to know how many pods
+belong to the group.
+
+These groups of Pods can be identified when they have:
+- the label `kueue.x-k8s.io/pod-group-name`, as a unique identifier for the group.
+- the annotation `kueue.x-k8s.io/pod-group-count`, the number of pods to expect in the group
+
+<<[UNRESOLVED expectations]>>
+
+1. Can we assume that when a Pod fails it won't be recreated?
+2. Alternatively, is there be a pod controller that handles recreation? If so, how do we know
+   if the job finished, so that there wouldn't be more recreations?
+
+Most likely, we should assume that the Pods are just plain Pods not controlled by a custom resource.
+Otherwise, users should prefer to integrate directly with the Workload API, which is not a
+significant effort, but would be more reliable.
```

Comment:
+1

Reply:
It makes sense. Maybe, there would be no pods unmanaged by anyone (CRDs or not CRDs) in the production.